### PR TITLE
provide a workaround for ruby/resolv/issues/23

### PR DESCRIPTION
**Pull request by BobSilent**
Hi @mburumaxwell,

this PR revives the changes that PR #369 brought with it and unfortunately were lost some time ago.

Can you please add this once again, until [Ruby `Resolv` bug](https://github.com/ruby/resolv/issues/23) gets merged.

Thanks a lot.

### CODE REVIEW

**Comment by mburumaxwell** [comment]
If I get this correctly, `WORKAROUND_CMD` will be an ENV variable?
What are some of the values that would be set?

**Comment by BobSilent** [comment]
Please see the details here #369

The issue I have is that within my network environment the docker host itself cannot resolve any network name, therefore some nameservers are configured, but unfortunately due to the bug (ruby/resolv/issues/23) I need to "patch" the `/etc/resolv.conf` before the image runs.
Currently I create an own docker image which is exactly the same as your images, with just one additional line during startup:
`sed -i 1' s/^/#/' /etc/resolv.conf`
in order to comment out the first nameserver entry which I am having issues with.


Currently I add an extra env with this value `sed -i 1' s/^/#/' /etc/resolv.conf`

**Comment by mburumaxwell** [comment]
Since you are building custom images, have you tried adding that to your Dockerfile with a custom `bin/run.sh`? For example:

```docker
ARG ECOSYSTEM
FROM ghcr.io/tinglesoftware/dependabot-updater-$ECOSYSTEM

WORKDIR ${DEPENDABOT_HOME}
COPY --chown=dependabot:dependabot run.sh bin/run.sh
```

**Comment by BobSilent** [comment]
That's what I'm doing right now, but to be honest I would really prefer to use your image out-of-the-box, without any modifications from my site. As this has the downside that I frequently have to update and maintain my images, if they do not work  with an updated extension.

That's why I really would like to ask you to add this minor and non breaking change again here - even if this is against your code cleanness (which I completely understand). 🙏

I'll continue to get the real fix PR merged @ruby/resolv - but unfortunately there I did not get any response up to now.

**Comment by Thulasi225** [comment]
That's why I really would like to ask you to add this minor and non breaking change again here - even if this is against your code cleanness (which I completely understand). 🙏

I'll continue to get the real fix PR merged @ruby/resolv - but unfortunately there I did not get any response up to now.

File: updater/bin/run.sh [hunk 1]
```diff
@@ -1,6 +1,11 @@
 #!/bin/bash
 set -e
 
+# WORKAROUND for https://github.com/ruby/resolv/issues/23, see also https://github.com/tinglesoftware/dependabot-azure-devops/pull/369
```
Comment (mburumaxwell) [inline updater/bin/run.sh:4]:
```diff
- # WORKAROUND for https://github.com/ruby/resolv/issues/23, see also https://github.com/tinglesoftware/dependabot-azure-devops/pull/369
+ # This is a WORKAROUND for https://github.com/ruby/resolv/issues/23
+ # see also https://github.com/tinglesoftware/dependabot-azure-devops/pull/369
+ # see also https://github.com/tinglesoftware/dependabot-azure-devops/pull/834

**Review by mburumaxwell — CHANGES_REQUESTED** [review]

**Comment by Thulasi225** [comment]
- # WORKAROUND for https://github.com/ruby/resolv/issues/23, see also https://github.com/tinglesoftware/dependabot-azure-devops/pull/369

File: updater/bin/run.sh (hunk 1 above)
Comment (BobSilent) [reply updater/bin/run.sh:4]: Sorry I missed the notification here from GitHub, will take care for this the next days!

Comment (mburumaxwell) [reply updater/bin/run.sh:4]: Okay. So this can be closed?

Comment (BobSilent) [reply updater/bin/run.sh]: if you mean merged yes, the issue is still open at `ruby/resolv`

**Review by mburumaxwell — APPROVED** [review]

**Pull request merged**